### PR TITLE
Make `SaltProviderNext` stable

### PR DESCRIPTION
**Pull request by origami-z**
- Deprecate all `UNSTABLE_` variables
- Enable QA stories for components listed #3536
    - Except ones in #3713

### CODE REVIEW

**Review by bhoppers2008 — APPROVED** [review]

File: packages/core/src/__tests__/__e2e__/salt-provider/SaltProvider.cy.tsx [hunk 1]
```diff
@@ -22,9 +22,14 @@ const TestComponent = ({
   const {
     theme,
     mode,
+    themeNext,
+    corner,
+    accent,
+    actionFont,
+    headingFont,
+    // Test backwards compatibilty using `UNSTABLE_` variables
     UNSTABLE_corner,
```
[the remark this answers is not in the document]
Comment (origami-z) [reply packages/core/src/__tests__/__e2e__/salt-provider/SaltProvider.cy.tsx:31]: Yes, making sure full backwards compatibility. I used these unstable variables in the sample app, so won't safe to assume no one used it before.

**Review by pseys — COMMENTED** [review]
In the Kitchen Sink examples all text links seem to have the wrong cursor behaviour on hover. Is this an issue within the code of the kitchen sink or an issue with Theme Next?

**Comment by origami-z** [comment]
> In the Kitchen Sink examples all text links seem to have the wrong cursor behaviour on hover. Is this an issue within the code of the kitchen sink or an issue with Theme Next?

Couldn't find a difference between 2 themes. What do you see?

**Comment by pseys** [comment]
> > In the Kitchen Sink examples all text links seem to have the wrong cursor behaviour on hover. Is this an issue within the code of the kitchen sink or an issue with Theme Next?
> 
> Couldn't find a difference between 2 themes. What do you see?

The link component shows a pointer whereas kitchen sink shows a text cursor.

**Review by joshwooding — APPROVED** [review]

**Review by pseys — APPROVED** [review]

**Pull request merged**